### DISCUSSION TITLE
Flags over warehouses depending on which walkers are allowed

### DISCUSSION THREAD

**Discussion by Ouaz** [Keriew/augustus#492] [Ideas]
Commit https://github.com/Keriew/augustus/commit/a1907408ab6bfbc47f60a15874d3563a568948f3 removed the "trade center" mechanic from warehouses. It makes totally sense, given the new warehouse options, but in the same time, it also removed the trade center flag, which was a nice graphical touch.

So, I don't know if it's possible, or, as always, if it would require too much work, but here's the suggestion: display a flag over a warehouse, depending on which walkers are allowed.

_________________________________
**->Only market ladies allowed**
![market_trade_check](https://user-images.githubusercontent.com/11946570/134764602-cca7044a-0168-49a1-a7db-4e2e1cd0079c.png)
![market_traders_flag](https://user-images.githubusercontent.com/11946570/134764621-9392ab2e-a388-486b-bbef-4a7e416745fb.gif)
_Market trading warehouse_

![flag_market_trade_a1](https://user-images.githubusercontent.com/11946570/134765231-5d0d7326-412d-49da-b122-03cb83938c78.png)[flag_market_trade_anim8.zip](https://github.com/Keriew/augustus/files/7229422/flag_market_trade_anim8.zip) (8 transparent png files, 30x24px, named a1 to a8)
_________________________________

**->Only caravans allowed**
![land_trade_check](https://user-images.githubusercontent.com/11946570/134764693-bf2c5f5a-6a44-490f-8216-4de6d7a84d47.png)
![land_traders_flag](https://user-images.githubusercontent.com/11946570/134764718-d1dca959-63f6-4d1a-aaea-6d38de48c8e5.gif) ![land_traders_flag2](https://user-images.githubusercontent.com/11946570/134764723-0dda6d75-21bd-473a-a2bc-a9f77a03511c.gif)
_Land trading warehouse_

![flag_land_trade_a1](https://user-images.githubusercontent.com/11946570/134765290-cbea7a49-164f-4269-9320-f7700bbc5e30.png)[flag_land_trade_anim8.zip](https://github.com/Keriew/augustus/files/7229428/flag_land_trade_anim8.zip) (8 transparent png files, 30x24px, named a1 to a8, camel icon with green flag) 
**- do not use this one, ![use the blue one instead, posted in this comment](https://github.com/Keriew/augustus/discussions/492#discussioncomment-1384336) -**
_________________________________

**->Only ships allowed**
![sea_trade_check](https://user-images.githubusercontent.com/11946570/134764761-ab446e82-da77-4ea1-8bd0-f29105695031.png)
![sea_traders_flag](https://user-images.githubusercontent.com/11946570/134764770-b6525d8b-f71e-4c88-aaa8-e20a47b9861d.gif)
_Sea trading warehouse_

![flag_sea_trade_a1](https://user-images.githubusercontent.com/11946570/134765306-0b69e0e2-1789-4ea6-b34c-11966f3a9ca5.png)[flag_sea_trade_anim8.zip](https://github.com/Keriew/augustus/files/7229430/flag_sea_trade_anim8.zip) (8 transparent png files, 30x24px, named a1 to a8)
_________________________________

**->Both caravans and ships allowed**
![both_land_sea_trade_check](https://user-images.githubusercontent.com/11946570/134764872-b6793d9f-fe5c-4be4-b804-3875b14a296f.png)
![both_land_sea_traders_flag](https://user-images.githubusercontent.com/11946570/134764877-67a04725-7466-4312-9825-245517ed4748.gif)
_Land and sea trading warehouse_ (C3 vanilla purple flag)

![flag_both_traders_a1](https://user-images.githubusercontent.com/11946570/134765345-d7a13491-7b3c-4f81-87a4-c343f2a7bb1c.png)[flag_both_traders_anim8.zip](https://github.com/Keriew/augustus/files/7229433/flag_both_traders_anim8.zip) (8 transparent png files, 30x24px, named a1 to a8)
(I post these too but they are in `c3.sg2` file in` Warehouse.bmp` section.)
_________________________________

**Why?**

- When having multiple warehouses in a big city, it's sometimes hard to remember who is allowed in which warehouse, and one may have to check each warehouse panel one by one. These flags shows directly on the map which walkers are allowed, without having to open the warehouse panels to verify.
- Nice graphical touch! :yum: 

Notes:

- "Halt resource to workshops or granaries" has no influence on which flag is displayed. For example, if only market ladies and resource transfer are allowed, the "market trading flag" is displayed.
- Any other combination than the ones listed above displays no flags (for example, market ladies + caravans, or all three, or nothing, etc.)
- Fun fact: in C3 vanilla, the trade center flag is not animated. The 7 other flag files are unused.
- The size of the flag images is 30x24px in C3 vanilla. I kept this size.
- I've set the delay between flag animations to 250ms, if this value is needed in the code.

Phew, done.

Let me know what you think about this (sorry, this is a huge request :smile: ), and if it's possible to implement this in Augustus.

Thanks in advance for the answer.

**Comment by crudelios** [comment]
I really like this. However, it doesn't factor in the market ladies + ship, market ladies + caravan or all combination.

How about, if only market ladies are accepted, the icon you provided is displayed.

Market ladies + caravan = the one you provided for caravan

Caravan only = the one you provided for caravan, Blue background

Market ladies + ships = the one you provided for ships, red background

All = the one for caravans and ships, red background

In essence, if the market ladies were accepted the flag's background would be red, otherwise it would be blue.

**Comment by Ouaz** [comment]
>  However, it doesn't factor in the market ladies + ship, market ladies + caravan or all combination.

So, you think it's better to have a flag for every possible combination?

Ok, I think I undertstand what you mean:

Colors are inherent to a walker:

Market ladies -> red color
Caravans -> green color
Ships -> blue color

So the remaining flags are:

Marked ladies + Caravans -> Camel icon on red background
Market ladies + Ships -> Ship icon on red background
All -> Trade center icon on red background
Nothing -> No flag

But I don't undertsand these two:

> Market ladies + caravan = the one you provided for caravan

Shouldn't it be Camel icon on red background?

> Caravan only = the one you provided for caravan, Blue background

Why the blue background? Caravan only is ![flag_land_trade_a1](https://user-images.githubusercontent.com/11946570/134765290-cbea7a49-164f-4269-9320-f7700bbc5e30.png) (Camel and green background).

Sorry if I'm a bit slow to understand. :p

Whatever, I will create the remaining flags this evening, and will post it here right after.

Thanks for your answer!

**Comment by crudelios** [comment]
Ah, I am sorry. I am color blind and only noticed two colors (blue and red), not the green.

**Comment by Ouaz** [comment]
Here's the remaining flags:

_____________________________
**->Market ladies and caravans allowed**

![market_land_trade_check](https://user-images.githubusercontent.com/11946570/134783743-946b0e24-8de8-4adb-a607-910d221e22cb.png)

![flag_market_land_trade_a1](https://user-images.githubusercontent.com/11946570/134783752-13c01e1f-6c89-4980-ae74-b33baff8367a.png)[flag_market_land_trade_anim8.zip](https://github.com/Keriew/augustus/files/7230247/flag_market_land_trade_anim8.zip) (8 transparent png files, 30x24px, named a1 to a8, camel icon with red background)

_____________________________
**->Market ladies and ships allowed**

![market_sea_trade_check](https://user-images.githubusercontent.com/11946570/134783807-beeda459-df2a-4369-91b4-a0234316c865.png)

![flag_market_sea_trade_a1](https://user-images.githubusercontent.com/11946570/134783819-da2dbf41-f63b-4228-8f94-5b83a05824f3.png)[flag_market_sea_trade_anim8.zip](https://github.com/Keriew/augustus/files/7230253/flag_market_sea_trade_anim8.zip) (8 transparent png files, 30x24px, named a1 to a8, ship icon with red background)

_____________________________
**->All traders allowed**

![all_traders_check](https://user-images.githubusercontent.com/11946570/134783911-544eec92-4997-4ac1-a948-dc755d79606b.png)

![flag_all_traders_a1](https://user-images.githubusercontent.com/11946570/134783926-d3728aef-fb79-4803-bb99-411bd12d3cad.png)[flag_all_traders_anim8.zip](https://github.com/Keriew/augustus/files/7230256/flag_all_traders_anim8.zip) (8 transparent png files, 30x24px, named a1 to a8, trade center icon with red background)

____________________________

Finally, thinking of it, it's better to have only 3 flag colors, like you said:

- Red flag for any combinations with market ladies allowed
- Blue flag for combinations with only caravans _**or**_ ships allowed
- Purple flag when only cararavans _**and**_ ships are allowed

I wasn't satisfied by the green flag anyway, because it didn't look good with the map grass in background (that's why the green flag is brighter than the other ones). And moreover, you can't see green, so that's the best reason to remove this color. :yum: 

So, when only caravans are allowed, use this flag instead, not the one in the opening post:

**->Only caravans allowed**

![land_trade_check](https://user-images.githubusercontent.com/11946570/134764693-bf2c5f5a-6a44-490f-8216-4de6d7a84d47.png)

![blue_flag_land_trade_a1](https://user-images.githubusercontent.com/11946570/134784474-6ffd88fb-dfed-4947-9646-24c73a204b2b.png)[blue_flag_land_trade_anim8.zip](https://github.com/Keriew/augustus/files/7230272/blue_flag_land_trade_anim8.zip) (8 transparent png files, 30x24px, named a1 to a8, camel icon with blue background)

________________________________

I think I haven't forgotten anything.

Let me know if you need something else.

**Reply by crudelios** [reply]
That looks good!

Now it's just the simple matter of implementing it 😉

**Comment by Keriew** [comment]
This looks like a really cool addition!

**Comment by Keriew** [comment]
Added in 8e29734
